File: src/main/java/CustomDocumentListener.java
import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;

public class CustomDocumentListener implements DocumentListener {

    private final Bulker bulker;

    public CustomDocumentListener(Bulker bulker) {
        this.bulker = bulker;
    }

    @Override
    public void insertUpdate(DocumentEvent e) {
        bulker.refreshOutputFileList();
    }

    @Override
    public void removeUpdate(DocumentEvent e) {
        bulker.refreshOutputFileList();
    }

    @Override
    public void changedUpdate(DocumentEvent e) {
        bulker.refreshOutputFileList();
    }
}

File: src/main/kotlin/CustomDocumentListener.kt
package de.griefed

import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

class CustomDocumentListener(private val bulker: BulkerKt) : DocumentListener {
    override fun insertUpdate(e: DocumentEvent?) {
        bulker.refreshOutputFileList()
    }

    override fun removeUpdate(e: DocumentEvent?) {
        bulker.refreshOutputFileList()
    }

    override fun changedUpdate(e: DocumentEvent?) {
        bulker.refreshOutputFileList()
    }
}
